scriptlets/install_checkbox_snaps

```diff
@@ -10,77 +10,107 @@
 # ready to run tests.
 # The `risk` parameter can be either "stable", "beta", "edge" and is used for
 #   both runtime and frontend
-# The `runtime_name` can be any of the runtime snaps (eg. checkbox16,
-#   checkbox24 etc.)
 # The `frontend_name` can be any of the frontend snaps (eg. checkbox,
 #   checkbox-tillamook).
 # The `frontend_track` parameter can be any valid track name (eg. 22.04, uc16).
 #   It is also used to appropriately install the snap (decide if a snap is
 #   classic or strict)
 
+
+get_ubuntu_arch() {
+    declare -A arch_map=([x86_64]=amd64 [aarch64]=arm64 [armv7l]=armhf [i686]=i386)
+    UNAME_ARCH=$(_run "uname -m")
+    echo "${arch_map[$UNAME_ARCH]:-$UNAME_ARCH}"
+}
+
+get_runtime() {
+    # get the store token from the device, if available
+    STORE=$(_run "snap model --assertion" | sed -n 's/^store:\s\(.*\)$/\1/p')
+    # get the snap info from the snap store and extract the base
+    BASE=$(curl -s -H 'Snap-Device-Series: 16' ${STORE:+ -H "Snap-Device-Store: $STORE"} "https://api.snapcraft.io/v2/snaps/info/$FRONTEND_NAME?fields=base" \
+    | jq -r \
+    --arg FRONTEND_TRACK $FRONTEND_TRACK --arg RISK $RISK --arg ARCH $(get_ubuntu_arch) \
+    '.["channel-map"] | .[] | select(.channel.track == $FRONTEND_TRACK and .channel.risk == $RISK and .channel.architecture == $ARCH) | .base')
```

Comment:
I would prefer this wasn't so complicated to read, why don't we use a temp var and put some spacing to the jq query, I really can't read it like this. Does this look good to you?
```
jq -r  \
  --arg FRONTEND_TRACK $FRONTEND_TRACK \
  --arg RISK $RISK --arg ARCH $(get_ubuntu_arch) \
    '
      .["channel-map"] | .[] | 
      select(
          .channel.track == $FRONTEND_TRACK and 
          .channel.risk == $RISK and
          .channel.architecture == $ARCH
      ) | 
      .base')
```